src/routes/CreateEvent.js

```diff
@@ -13,42 +15,172 @@ const CreateContainer = styled('div')`
+  margin: 40px auto 140px auto;
+  font-size: 14px;
+`
+
+const UnlockedLogo = styled('a')`
+  color: #ff6771;
+  font-weight: 800;
+`
+
+const AddrDisplay = styled('span')`
+  background-color: #6e76ff;
+  color: #fff;
+  border-radius: 2px;
+  padding: 5px;
+  font-size: 10px;
+  margin-right: 10px;
+`
+
+const UnlockUserDetails = styled('div')`
+  display: flex;
+  flex-direction: row;
+  margin-bottom: 30px;
+  transition: transform 300ms ease-in-out;
+`
 
-  _onCreated = ({ id }, deposit, limitOfParticipants, coolingPeriod) => {
-    this.props.history.push(
+export default function Create() {
+  const [password, setPassword] = useState('')
+  const [locked, setLocked] = useState('pending')
+  const [membership, setMembership] = useState('')
+  const [membershipAddr, setMembershipAddr] = useState('')
+  let history = useHistory()
+
+  const _onCreated = ({ id }, deposit, limitOfParticipants, coolingPeriod) => {
+    history.push(
       `/deploy?id=${id}&deposit=${deposit}&limitOfParticipants=${limitOfParticipants}&coolingPeriod=${coolingPeriod}`
     )
   }
-}
 
-export default withRouter(Create)
+  const updateUnlockUser = () => {
+    // sets membership lock and lock name to state
+    let unlock = window.unlockProtocol.blockchainData()
+    let locks = Object.keys(unlock.locks).map(i => unlock.locks[i])
+    let bronze = locks.find(o => o.name.includes('Kickback Bronze'))
+    let gold = locks.find(o => o.name.includes('Kickback Gold'))
+    if (gold) {
+      setMembership(gold.name)
+      setMembershipAddr(gold.address)
+    } else {
+      setMembership(bronze.name)
+      setMembershipAddr(bronze.address)
+    }
+  }
+
+  const unlockHandler = async e => {
+    /*
+      Status can either be 'unlocked' or 'locked'...
+      If state is 'unlocked': implement code here which will be triggered when 
+      the current visitor has a valid lock key  
+      If state is 'locked': implement code here which will be
+      triggered when the current visitor does not have a valid lock key
+    */
+
+    setLocked(e.detail)
+
+    // run this loop only if unlocked
+    if (e.detail === 'unlocked') {
+      // blockchainData() will load empty first, check if loaded before updating state, checks every 100 ms
+      let checkExist = setInterval(function() {
+        if (window.unlockProtocol.blockchainData()) {
+          // once loaded, update hooks with lock info, and exit timer loop
+          updateUnlockUser()
+          clearInterval(checkExist)
+        }
+      }, 100)
+    }
+  }
+
+  const checkout = () => {
+    window.ethereum.enable()
+    window.unlockProtocol && window.unlockProtocol.loadCheckoutModal()
+  }
+
+  useEffect(() => {
+    // don't load paywall if testing with ganache
+    if (ENV !== 'local') {
+      window.addEventListener('unlockProtocol', unlockHandler)
+    } else {
+      setLocked('unlocked')
+    }
+  }, [])
+
+  useEffect(() => {
+    // remove unlock in component unmount1
+    return () => {
```

Comment:
Can this not be bundled with the other useEffect's return statement?

Reply:
You're right. dbc56b9